Bottom: 7f096cd9df96083d1d3cfad1eb065d0c3389ab8e
Top:    c4906b969c5b4cb5a5581ea26ac771988a8df770
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-09-05 17:41:14 +0100

hid/gtk: Use the g_timeout_add() function, not the deprecated GTK one.


---

diff --git a/src/hid/gtk/gtkhid-main.c b/src/hid/gtk/gtkhid-main.c
index e201d3c..89d50be 100644
--- a/src/hid/gtk/gtkhid-main.c
+++ b/src/hid/gtk/gtkhid-main.c
@@ -372,7 +372,7 @@ ghid_set_crosshair (int x, int y, int action)
 typedef struct
 {
   void (*func) (hidval);
-  gint id;
+  guint id;
   hidval user_data;
 }
 GuiTimer;
@@ -397,7 +397,7 @@ ghid_add_timer (void (*func) (hidval user_data),
 
   timer->func = func;
   timer->user_data = user_data;
-  timer->id = gtk_timeout_add (milliseconds, (GtkFunction) ghid_timer, timer);
+  timer->id = g_timeout_add (milliseconds, (GSourceFunc) ghid_timer, timer);
   ret.ptr = (void *) timer;
   return ret;
 }
@@ -407,7 +407,7 @@ ghid_stop_timer (hidval timer)
 {
   void *ptr = timer.ptr;
 
-  gtk_timeout_remove (((GuiTimer *) ptr)->id);
+  g_source_remove (((GuiTimer *) ptr)->id);
   g_free( ptr );
 }
